### PR TITLE
Add type hints to arelle.Locale

### PR DESCRIPTION
**Pull request by selgamal**
#### Reason for change
Add type hints to arelle.Locale

#### Description of change
Following are type ignores in this pr:
[L49](https://github.com/Arelle/Arelle/blob/adf7c8e327941fd097ba79b381a4df9bc5a2c4d1/arelle/Locale.py#L49) # type: ignore[name-defined] # builtins
[L66](https://github.com/Arelle/Arelle/blob/adf7c8e327941fd097ba79b381a4df9bc5a2c4d1/arelle/Locale.py#L66) # type: ignore[name-defined] # builtins
[L534](https://github.com/Arelle/Arelle/blob/adf7c8e327941fd097ba79b381a4df9bc5a2c4d1/arelle/Locale.py#L534) # type: ignore[name-defined] # builtins
[L550](https://github.com/Arelle/Arelle/blob/adf7c8e327941fd097ba79b381a4df9bc5a2c4d1/arelle/Locale.py#L550) # type: ignore[name-defined] # builtins

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by sagesmith-wf — DISMISSED** [review]

File: arelle/Locale.py
Comment (sagesmith-wf) [inline arelle/Locale.py:72]: Nit: extra whitespace

**Review by sagesmith-wf — APPROVED** [review]

File: arelle/Locale.py
Comment (selgamal) [reply arelle/Locale.py:552]: The problem here is that Decimal does not expect a Fraction, so I have to exclude the fraction anyway and isolate its test.

File: arelle/Locale.py
Comment (selgamal) [reply arelle/Locale.py:86]: language_code can be None, we have to ensure fallback is returned here.

File: arelle/Locale.py
Comment (austinmatherne-wk) [inline arelle/Locale.py:84]: It isn't conventional for python, but arelle uses camel case everywhere, we should stay consistent:
```suggestion
    languageCode, encoding = locale.getdefaultlocale()
```

File: arelle/Locale.py
Comment (austinmatherne-wk) [reply arelle/Locale.py:86]: I think Sage meant you could do this, which handles the `None` case but is a little more succinct:

```python
    if isinstance(language_code, str):
        return language_code.replace("_", "-")
    return "en"
```

File: arelle/Locale.py
Comment (austinmatherne-wk) [reply arelle/Locale.py:552]: This is an annoying limitation of the type checker. It would be better to update the condition to explicitly check for what we want:
```python
    elif isinstance(value, (str, int)):
        value = Decimal(value)
    elif isinstance(value, Fraction):
        value = Decimal(float(value))
```

**Review by austinmatherne-wk — APPROVED** [review]

File: tests/unit_tests/arelle/test_locale.py
Comment (sagesmith-wf) [reply tests/unit_tests/arelle/test_locale.py:64]: Yeah agreed... rad! Really appreciate it!

File: arelle/typing.py
Comment (selgamal) [reply arelle/typing.py:34]: @austinmatherne-wk 
Actually I should have removed the comments, it was just a reminder while working, I will keep them for now, let me know if I should remove them.

File: tests/unit_tests/arelle/test_locale.py
Comment (selgamal) [reply tests/unit_tests/arelle/test_locale.py:64]: Thank you for the review! hope I am not giving you too much trouble.

File: arelle/typing.py
Comment (austinmatherne-wk) [reply arelle/typing.py]: They're useful. Feel free to keep them.

**Review by austinmatherne-wk — DISMISSED** [review]

**Comment by austinmatherne-wk** [comment]
This PR looks good if you want to move it out of draft status

File: tests/unit_tests/arelle/test_locale.py
Comment (austinmatherne-wk) [reply tests/unit_tests/arelle/test_locale.py:64]: Not at all. We're very thankful for the contributions!

**Review by sagesmith-wf — DISMISSED** [review]

File: pyproject.toml
Comment (austinmatherne-wk) [inline pyproject.toml:109]: nit: sort is off

**Review by austinmatherne-wk — APPROVED** [review]

File: pyproject.toml
Comment (selgamal) [reply pyproject.toml:109]: @austinmatherne-wk
I am waiting on #342 to merge to resolve any other conflicts here in one go.

Comment (selgamal) [reply pyproject.toml]: @austinmatherne-wk 
aside note, seems that the conformance suites are running on ubuntu only, is that the intention?

Comment (austinmatherne-wk) [reply pyproject.toml]: Yes. We considered running on multiple platforms, but decided against it due to the increased execution time.

**Review by austinmatherne-wk — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Pull request merged**
